### PR TITLE
Adds implementation of DialNumber RPC

### PR DESCRIPTION
**Pull request by dev-gh**
DialNumber allows to initiate call from user's device trough device attached to the system.

All characters shall be stripped from string except digits 0-9 and * # , ; (SDL must validate the "number" param of DialNumber RPC received from mobile app and strip all characters from string except digits 0-9 and * # , ; when transferring this DialNumber RPC to HMI).

Implements: APPLINK-11266

### CODE REVIEW

**Comment by dev-gh** [comment]
@AGaliuzov @OHerasym @AByzhynar @anosach-luxoft @VProdanov 
Please review

**Comment by AGaliuzov** [comment]
@dev-gh looks good to me.
Reviewed

**Comment by VProdanov** [comment]
Reviewed

**Comment by OHerasym** [comment]
Reviewed

File: src/components/application_manager/include/application_manager/commands/hmi/dial_number_request.h [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  /**
+   * @brief DialNumberRequest class destructor
+   **/
+  virtual ~DialNumberRequest();
+
+  /**
+   * @brief Execute command
+   **/
+  virtual void Run();
```
Comment (AByzhynar) [inline src/components/application_manager/include/application_manager/commands/hmi/dial_number_request.h]: @dev-gh Add OVERRIDE please

File: src/components/application_manager/include/application_manager/smart_object_keys.h [hunk 2]
```diff
@@ -280,6 +280,7 @@ const char location_name[] = "locationName";
 const char location_description[] = "locationDescription";
 const char address_lines[] = "addressLines";
 const char phone_number[] = "phoneNumber";
+const char number[] = "number";
```
Comment (AByzhynar) [inline src/components/application_manager/include/application_manager/smart_object_keys.h]: @dev-gh We have 2 numbers: `phone_number` & just `number`. Do we need them both? What is the difference between them?

**Comment by AByzhynar** [comment]
Reviewed

File: src/components/application_manager/include/application_manager/smart_object_keys.h (hunk 2 above)
Comment (dev-gh) [inline src/components/application_manager/include/application_manager/smart_object_keys.h]: @AByzhynar 
As I found out 'phone_number' is being used in SendLocation request according to its API.
New field 'number' is necessary for DialNumber request.

File: src/components/application_manager/include/application_manager/commands/hmi/dial_number_request.h (hunk 1 above)
Comment (dev-gh) [inline src/components/application_manager/include/application_manager/commands/hmi/dial_number_request.h]: @AByzhynar 
Fixed with 2013219874fc227221a53abf3bfe9b3368fa5d93

File: src/components/application_manager/include/application_manager/commands/mobile/dial_number_response.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+   * @brief DialNumberResponse class constructor
+   *
+   * @param message Incoming SmartObject message
+   **/
+  explicit DialNumberResponse(const MessageSharedPtr& message);
+
+  /**
+   * @brief DialNumberResponse class destructor
+   **/
+  virtual ~DialNumberResponse();
```
Comment (AGaliuzov) [inline src/components/application_manager/include/application_manager/commands/mobile/dial_number_response.h]: `OVERRIDE` insead of `virtual`

Comment (dev-gh) [inline src/components/application_manager/include/application_manager/commands/mobile/dial_number_response.h]: @AGaliuzov 
Done with 85172a5531462dc0ed7e923859faad5471dfb73e

**Comment by AGaliuzov** [comment]
Reviewed

**Pull request merged**